Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -88,24 +87,23 @@ namespace RGL
 
     void LidarRaycaster::ConfigureRaycastResultFlags(ROS2::RaycastResultFlags flags)
     {
-        m_resultFlags = flags;
         m_rglRaycastResults.m_fields.clear();
-        m_rglRaycastResults.m_isHit.clear();
         m_rglRaycastResults.m_xyz.clear();
         m_rglRaycastResults.m_distance.clear();
 
-        if ((flags & ROS2::RaycastResultFlags::Points) == ROS2::RaycastResultFlags::Points)
+        m_raycastResults = ROS2::RaycastResults(flags);
+
+        if (ROS2::IsFlagEnabled<ROS2::RaycastResultFlags::Point>(flags))
         {
-            m_rglRaycastResults.m_fields.push_back(RGL_FIELD_IS_HIT_I32);
```

Comment:
I see that this is not used in results, however pipeline graph contains compaction on this field - does it still work correctly?

Reply:
To the best of my knowledge it is not required to specify this field in the yield nodes for RGL to work correctly internally. @msz-rai might be a better person to ask however.

Reply:
@alek-kam-robotec-ai you are right. Compact node will require this field internally and the user does not have to request it additionally.